src/ui/link/index.tsx

```diff
@@ -72,6 +72,16 @@ const Link = forwardRef<HTMLButtonElement | HTMLAnchorElement, LinkProps>(
             )) ||
           ''
         : to;
+    const hrefDestination =
+      href != null
+        ? href
+        : typeof to !== 'string'
+        ? (route &&
+            createPath(
+              generateLocationFromPath(route.path, routeAttributes)
+            )) ||
+          ''
+        : `${routeAttributes.basePath}${to}`;
```

Comment:
yeah so this is obviously hacked in, looking for input on a better way of handling this.

Essentially the problem here is that `linkDestination` is used for both router actions and anchor props (href). Router actions automatically prepends basePath whilst anchor props do not, hence why we are in the current state.

This example splits the logic so they are handled separately. The approach is very naive, please provide suggestions. Alternatively, we could look at converting string typed `to`'s to routes?

Anyways, looking for feedback, thanks.

Reply:
We already calculate the href above, don't need to recalculate everything again. We just need to understand if we need to add basepath.

```js
const staticBasePath = href == null && typeof to === 'string' ? routeAttributes.basePath : ''
// ...
validLinkType,
  {
  ...rest,
  href: staticBasePath + linkDestination,
```

Reply:
I've applied these changes, thanks @albertogasparin